EventbriteApiV3/EventbriteContext.cs

```diff
@@ -21,15 +22,32 @@ public EventsSearchApiResponse GetEvents(BaseSearchCriterias searchCriterias)
         {
             return new EventSearchEventbriteRequest(this, searchCriterias).GetResponse();
         }
+        private async Task FillDescriptions(BaseSearchCriterias searchCriterias, IList<Event> events)
+        {
+            if (searchCriterias.RetrieveFullDescription)
+            {
+                var tasks = events.Select(x => new EventDescriptionRequest(this, x.Id).GetResponseAsync()).ToArray();
+                await Task.WhenAll(tasks);
+                var descriptions = tasks.Select(async x => (await x).Description).ToArray();
+                for (var i = 0; i < events.Count; i++)
+                {
+                    events[i].LongDescription = new Model.TextHtmlString { Html = await descriptions[i] };
```

Comment:
This will wait for requests to execute one after the other. You could execute them all simultanously with the method: Task.WhenAll(TaskList.ToArray());

ex`
var tasks = events.select(x => new EventDescriptionRequest(this, x.Id)).GetResponseAsync()).ToArray();
await Task.WhenAll(tasks);
var descriptions = tasks.Select(async x => (await x).Description).ToArray();
for(var i = 0; i<=events.Count; i++) {
events[i].LongDescription = new Model.TextHtmlString { Html = descriptions[i] }
}

`

Reply:
Bro, I do not agree.

 var tasks = events.Select(x => new EventDescriptionRequest(this, x.Id).GetResponseAsync()).ToArray();
                await Task.WhenAll(tasks);

tasks is an array and all is done asynchronously.
Now there are more 'awaits' there, but the task already has executed. Problem here is not async but the 'beauty' of the multiple awaits after 'WhenAll'. I Cannot refactor the awaits away. Sorry.